### PR TITLE
fix: syncthing restarts with relay enabled

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier Lopez <javier@okteto.com>

# Proposed changes

Fixes #DEV-395

- Adds a new volume as a source of truth for the syncthing config which is separated from the syncthing data
- Bump okteto binary to use these new volumes

Depends on:
- https://github.com/okteto/supervisor/pull/15
- https://github.com/okteto/bin-image/pull/12

## How to validate

1. Run okteto up 
1. Remove /var/syncthing/config.xml
1. Kill syncthing process 
1. Check that the /var/syncthing/config.xml is regenerated with the relay enabled to false

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: pkg/model/dev.go
Comment (andreafalzetti) [inline pkg/model/dev.go:46]:
```suggestion
	OktetoBinImageTag = "okteto/bin:1.6.0"
```

Comment (andreafalzetti) [reply pkg/model/dev.go:46]: reminder to change this before merging

File: pkg/k8s/apps/translate.go
Comment (andreafalzetti) [inline pkg/k8s/apps/translate.go:592]: @jLopezbarb the unit test `Test_translateWithVolumes` needs to be updated reflecting these changes

**Review by andreafalzetti — APPROVED** [review]
I was able to reproduce the bug and validate the fix, nice work.

Approving with comments:

1. fix the tests
2. add PR label
3. change `OktetoBinImageTag`

File: pkg/model/dev.go
Comment (ifbyol) [inline pkg/model/dev.go:46]: Remember to change it to the good image name

**Review by ifbyol — COMMENTED** [review]
If someone was disconnected from okteto up, and executes okteto up, will this work fine?

**Comment by jLopezbarb** [comment]
> If someone was disconnected from okteto up, and executes okteto up, will this work fine?

What do you mean by disconnected? Does an exit works? Everytime that we start an `okteto up` session we copy the files that could be removed so it'll work for that case

**Comment by ifbyol** [comment]
> > If someone was disconnected from okteto up, and executes okteto up, will this work fine?
> 
> What do you mean by disconnected? Does an exit works? Everytime that we start an `okteto up` session we copy the files that could be removed so it'll work for that case

I mean any disconnection from the pod which doesn't imply to delete the dev container. `exit` would be the same case I guess

**Review by andreafalzetti — APPROVED** [review]

**Review by ifbyol — APPROVED** [review]

**Pull request merged**
